centinel/experiments/baseline.py

```diff
@@ -0,0 +1,271 @@
+        pcap_results = {}
+        pcap_indexes = {}
+        url_index = 0
+        comments = ""
+
+        # we may want to make this threaded and concurrent
+        for line in file_contents:
+            line = line.strip()
+
+            # parse file comments, if it looks like "key : value",
+            # parse it as a key-value pair. otherwise, just
+            # store it as a raw comment.
+            if line[0] == '#':
+                line = line[1:].strip()
+                if len(line.split(':')) > 1:
+                    key, value = line.split(':', 1)
+                    key = key.strip()
+                    value = value.strip()
+                    file_metadata[key] = value
+                else:
+                    file_comments.append(line)
+                continue
+
+            url = line
+            meta = ''
+            url_index = url_index + 1
+
+            # handle cases where URL is enclosed in quotes
+            if line[0] == '"':
+                next_quote = line[1:].find('"')
+                url = line[1:next_quote + 1]
+                # try to separate metadata, if any
+                try:
+                    meta = ','.join(line[next_quote + 2:].split(',')[1:])
+                except:
+                    pass
+
+            # if the line doesn't start with a quote
+            # but the list entry has comma separated meta-data
+            elif len(line.split(',')) > 1:
+                url, meta = line.split(',', 1)
+                # remove trailing spaces
+                url = url.strip()
+
+            http_ssl = False
+            ssl_port = 443
+            http_path = '/'
+
+            # parse the URL to extract netlocation, HTTP path, domain name,
+            # and HTTP method (SSL or plain)
+            try:
+                http_netloc = ''.join(urlparse.urlparse(url).netloc)
+
+                # if netloc is not urlparse-able, add // to the start
+                # of URL
+                if http_netloc == '':
+                    url = '//' + url
+                    http_netloc = ''.join(urlparse.urlparse(url).netloc)
+
+                http_path = urlparse.urlparse(url).path
+                if http_path == '':
+                    http_path = '/'
+
+                # we assume scheme is either empty, or "http", or "https"
+                # other schemes (e.g. "ftp") are out of the scope of this
+                # measuremnt
+                if urlparse.urlparse(url).scheme == "https":
+                    http_ssl = True
+                    if len(http_netloc.split(':')) == 2:
+                        ssl_port = http_netloc.split(':')[1]
+
+            except Exception as exp:
+                logging.warning("%s: failed to parse URL: %s" %(url, str(exp)))
+                http_netloc = url
+                http_ssl    = False
+                ssl_port = 443
+                http_path   = '/'
+
+            domain_name = http_netloc.split(':')[0]
```

Comment:
is this going to work if you get an exception? Wouldn't that just give you the scheme, e.g. http, if you hit an exception? Also, you should just put this where you put the other URL parsing stuff or in the exception handling stuff